services/QuillLMS/engines/evidence/app/models/evidence/research/gen_ai/llm_config.rb

```diff
@@ -14,8 +14,21 @@ module Evidence
   module Research
     module GenAI
       class LLMConfig < ApplicationRecord
+        VENDORS = [
+          GOOGLE = 'google'
+        ]
+
         validates :vendor, presence: true
         validates :version, presence: true
+
+        attr_readonly :vendor, :version
+
+        def llm_client
```

Comment:
Consider converting this to a hash lookup instead of a case statement.
```ruby
VENDOR_MAP = {
  GOOGLE => Evidence::Gemini::Completion
}
def llm_client = VENDOR_MAP.fetch(vendor) { raise UnsupportedVendorError }